lib/ansible/modules/packaging/os/apt_repository.py

```diff
@@ -503,6 +503,9 @@ def main():
         else:
             module.fail_json(msg='%s is not installed, and install_python_apt is False' % PYTHON_APT)
 
+    if repo is None:
```

Comment:
What about empty string? I'd modify clause to be `if not repo:` to catch all falsy values.